src/common/selectors/selectorUtils.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2023, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { Messages, Connection } from '@salesforce/core';
+import { QueryResult, Record } from 'jsforce';
+
+Messages.importMessagesDirectory(__dirname);
+const messages = Messages.loadMessages('@salesforce/plugin-devops-center', 'commonErrors');
+
+export async function runSafeQuery<T extends Record>(
+  con: Connection,
+  queryStr: string,
+  allowNullResults = false
+): Promise<QueryResult<T>> {
+  // Parameters verification
+  if (!con) {
+    throw messages.createError('error.connection-required');
+  }
+  if (typeof queryStr !== 'string' || queryStr.trim().length === 0) {
+    throw messages.createError('error.query-string-required');
+  }
```

Comment:
@npiccolo Typescript knows these are not undefined and of the correct type. Why the extra checks?

Reply:
The connection check I did it because we can have potentially things like this:
`this.flags['devops-center-username'] as Org).getConnection()`, this can't fail at some point and pass something wrong to the method ???
And the queryString I tested because I have already tested the other parameter